### PR TITLE
[doc]Optimize the installation steps of the pulsar library of the Go client

### PR DESCRIPTION
**Pull request by zhaoyajun2009**
**Motivation**
Optimize the installation steps of the pulsar library of the Go client

**Modifications**
"
### Install go package

You can install the `pulsar` library locally using `go get` download to local or downloading online.  

Download to local:

```bash
$ go get -u "github.com/apache/pulsar-client-go/pulsar"
```
If you can't download, please try it by configure [CDN accelerated proxy]( https://learnku.com/go/wikis/38122).

Once installed locally, you can import it into your project:

```go
import "github.com/apache/pulsar-client-go/pulsar"
```

Online download(for example):

```bash
$ mkdir test_dir && cd test_dir 
```
Write a sample script in 'test_dir' directory , such as ‘test_example.go’,and write ‘package main’ at the beginning of the file.

```bash
$ go mod init test_dir 
$ go mod tidy && go mod download
$ go build test_example.go
$ ./test_example
```
"

**### Documentation**
  
- [x] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@wolfstudy can you review this PR from the technical perspective? Then writers can review it from the technical writing perspective, thanks.

**Comment by zhaoyajun2009** [comment]
@freeznet can you review this PR?

File: site2/docs/client-libraries-go.md [hunk 1]
```diff
@@ -16,18 +16,35 @@ You can use Pulsar [Go client](https://github.com/apache/pulsar-client-go) to cr
 
 ### Install go package
 
-You can install the `pulsar` library locally using `go get`.  
+You can install the `pulsar` library locally using `go get` download to local or downloading online.
```
Comment (freeznet) [inline site2/docs/client-libraries-go.md:19]:
```suggestion
You can get the `pulsar` library by using `go get` or use it with `go module`.  
```

File: site2/docs/client-libraries-go.md [hunk 2]
```diff
@@ -16,18 +16,35 @@ You can use Pulsar [Go client](https://github.com/apache/pulsar-client-go) to cr
...
 
-You can install the `pulsar` library locally using `go get`.  
+You can install the `pulsar` library locally using `go get` download to local or downloading online.  
+
+Download to local:
 
 ```bash
 $ go get -u "github.com/apache/pulsar-client-go/pulsar"
 ```
+If you can't download, please try it by configure [CDN accelerated proxy]( https://learnku.com/go/wikis/38122).
```
Comment (freeznet) [inline site2/docs/client-libraries-go.md:26]: this should only applied to China users due the network issue, so i dont think it is proper to put it here.

File: site2/docs/client-libraries-go.md [hunk 3]
```diff
@@ -16,18 +16,35 @@ You can use Pulsar [Go client](https://github.com/apache/pulsar-client-go) to cr
...
 
 Once installed locally, you can import it into your project:
 
 ```go
 import "github.com/apache/pulsar-client-go/pulsar"
 ```
 
+Online download(for example):
```
Comment (freeznet) [inline site2/docs/client-libraries-go.md:34]:
```suggestion
Use with go module:
```

**Review by freeznet — COMMENTED** [review]
lgtm, left some comments.

**Review by freeznet — APPROVED** [review]

File: site2/docs/client-libraries-go.md [hunk 4]
```diff
@@ -28,6 +30,20 @@ Once installed locally, you can import it into your project:
 import "github.com/apache/pulsar-client-go/pulsar"
 ```
 
+Use with go module:
+
+```bash
+$ mkdir test_dir && cd test_dir 
+```
+Write a sample script in 'test_dir' directory , such as ‘test_example.go’,and write ‘package main’ at the beginning of the file.
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-go.md:38]:
```suggestion
Write a sample script in the `test_dir` directory (such as `test_example.go`) and write `package main` at the beginning of the file.
```

File: site2/docs/client-libraries-go.md [hunk 5]
```diff
@@ -16,7 +16,9 @@ You can use Pulsar [Go client](https://github.com/apache/pulsar-client-go) to cr
 
 ### Install go package
 
-You can install the `pulsar` library locally using `go get`.  
+You can get the `pulsar` library by using `go get` or use it with `go module`.  
+
+Download to local:
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-go.md:21]:
```suggestion
Download Go client to your local environment:
```

Comment (zhaoyajun2009) [inline site2/docs/client-libraries-go.md:21]: Download library of Go client to your local environment:

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**